src/js/tasks/osSiloPrice.js

```diff
@@ -145,6 +150,182 @@ const calculateMinBuyingPrice = (lendingAPY) => {
     return minPriceScaled > minAllowed ? minPriceScaled : minAllowed;
 };
 
+/**
+ * @notice Estimates the average withdrawal time for a given ARM contract.
+ * @dev This function calculates the estimated withdrawal time based on the availability of tokens
+ *      in the OS Vault and ARM contract, as well as the status of withdrawal requests.
+ * 
+ * Scenarios:
+ * 1. If there is more wS available in the OS Vault than oS in the ARM contract, the withdrawal time
+ *    is estimated to be close to 1 day.
+ * 2. If there are enough unclaimed withdrawal requests older than 13 days to cover the required amount,
+ *    the withdrawal time is estimated to be close to 1 day.
+ * 3. If there are not enough unclaimed withdrawal requests (both older and younger than 13 days) to cover
+ *    the required amount, the withdrawal time is estimated to be the maximum of 14 days.
+ * 4. If there are enough unclaimed withdrawal requests (including recent ones) to cover the required amount,
+ *    the withdrawal time is calculated using a weighted average based on the age of the requests.
+ */
+const estimateAverageWithdrawTime = async (arm, block) => {
+    const blockTag = await getBlock(block);
+    const timestamp = await hre.ethers.provider.getBlock(blockTag).then(b => b.timestamp);
+    log(`Using block number: ${blockTag} at timestamp: ${timestamp}`);
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////
+    /// --- Fetching wS holding from OS Vault
+    ////////////////////////////////////////////////////////////////////////////////////////////////////
+    log(`\nFetching WS data from OS Vault ...`);
+    let wSAvailable = 0n;
+    const wSAddress = await arm.token0({ blockTag });
+    const wS = await hre.ethers.getContractAt(
+        [`function balanceOf(address owner) external view returns (uint256)`],
+        wSAddress
+    );
+    wSAvailable += await wS.balanceOf(await arm.vault({ blockTag }), { blockTag });
+    log(`ws balanceOf OSVault    : ${formatUnits(wSAvailable, 18)}`);
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////
+    /// --- Fetching data from OS Vault
+    ////////////////////////////////////////////////////////////////////////////////////////////////////
+    const vaultAddress = await arm.vault({ blockTag });
```

Comment:
nit: `arm.vault` and `blockTag` again

Reply:
Fixed in this commit: [6d4b65e](https://github.com/OriginProtocol/arm-oeth/pull/124/commits/6d4b65e948d4e37651653beaf08a53a3adbc0828)